0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,415 @@
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Overview
+
+This specification codifies a common interface for a **settlement engine**. Settlement engines are services which perform two primary operations:
+
+1. Send payments: execute outgoing settlements
+2. Receive payments: acknowledge incoming settlements
+
+Counterparties may operate compatible settlement engines to settle their liabilities between one another. Different implementations may utilize different settlement systems or types of settlements, such as:
+
+- Sending money or assets to the counterparty
+- Transferring value on a shared ledger
+- Signing and exchanging payment channel claims
+- Performing a task for the counterparty with a mutually ascribed value
+
+## Usage in Interledger
+
+### Interledger packet flow
+
+In the [Interledger protocol](../0001-interledger-architecture/0001-interledger-architecture.md), a **peer** is a connector who is a counterparty to another connector. Connectors clear and fulfill ILP packets with their peers, which represent conditional IOUs that affect their mutual accounting relationships.
+
+A connector may extend a given peer a limited line of credit, or none at all, depending upon their trustworthiness. As the connector receives incoming ILP PREPARE packets from a peer, forwards them, and returns corresponding ILP FULFILL packets, that peer's liabilities to the connector accrue. If the peer's liabilities exceed the credit limit assigned to it, the connector may reject and decline to forward additional packets.
+
+In order to continue transacting, the peer must settle their liabilities. In most cases, this is accomplished through sending a payment on a settlement system that both peers have agreed to use. The connector should acknowledge the incoming payment&mdash;irrevocably discharging some liability the peer owed to it&mdash;and enabling it to clear and forward subsequent packets from the peer on credit.
+
+Settlement engines provide a standardized mechanism for Interledger connectors to coordinate and reconcile these settlements.
+
+### Accounting
+
+An **account** represents a record of transactions between counterparties, denominated in some fungible asset. Each account may be comprised of multiple account balances, which each represent the net difference between amounts received (credits) and amounts owed (debits) for some subset of their transactions.
+
+Interledger connectors are RECOMMENDED to operate an **accounting system** which keeps a record of two balances for each account:
+
+- **Accounts payable**, the amount owed by the connector to its peer for packets its peer has fulfilled.
+  - Positive amount indicates the connector is indebted to its peer (a _liability_ for the connector).
+  - Negative amount indicates the connector has prefunded.
```

Comment:
> Negative amount indicates the connector has prefunded.

The above is unclear to me. How about: 

```suggestion
  - Negative amount indicates the connector has prefunded its account with its peer, typically by sending actual value on an unerlying ledger.
```

Reply:
How about "Negative amount indicates the connector has sent a prepayment to its peer and has as much credit available to spend"?

FWIW, I don't think we should equate a negative payable with a positive receivable. The _net_ amount owed is equivalent, yes, but the two represent different reasons for why that net amount is what it is.

Reply:
> How about "Negative amount indicates the connector has sent a prepayment to its peer and has as much credit available to spend"?

Hmmm...how about just: "Negative amount indicates the connector has sent a pre-payment to its peer"?

Reply:
> FWIW, I don't think we should equate a negative payable with a positive receivable. The net amount owed is equivalent, yes, but the two represent different reasons for why that net amount is what it is.

Maybe we're saying the same thing: I think we both agree that these should be two different accounts (a payable and a receivable). But from a pure accounting perspective, a negative payable is a receivable and vice-versa.

That said, it's perhaps not valuable to mention that here, so I'm fine striking that last bit if you want.